replicators/src/mysql_connector/connector.rs

```diff
@@ -128,8 +132,27 @@ impl MySqlBinlogConnector {
         };
 
         connector.register_as_replica().await?;
-        connector.request_binlog().await?;
-
+        let binlog_request = connector.request_binlog().await;
+        match binlog_request {
+            Ok(()) => (),
+            Err(mysql_async::Error::Server(ref err)) if err.code == 1236 => {
```

Comment:
is there a constant somewhere (maybe in [mysql_common](https://github.com/blackbeam/rust_mysql_common)) we can pull this error code from?

if not, I think it's worth making it a top-level constant since it's currently a bit of a magic number

Reply:
Indeed plain number looks bad. Added `ER_MASTER_FATAL_ERROR_READING_BINLOG` from `mysql_srv::ErrorKind`